File: src/main/java/com/whirlpool/paintedplanks/item/ModItems.java
package com.whirlpool.paintedplanks.item;

import com.whirlpool.paintedplanks.PaintedPlanks;
import net.minecraft.world.item.Item;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import net.minecraftforge.registries.RegistryObject;

public class ModItems {
    public static final DeferredRegister<Item> ITEMS =
            DeferredRegister.create(ForgeRegistries.ITEMS, PaintedPlanks.MODID);

    public static final RegistryObject<Item> TEST_ITEM = ITEMS.register("test_item",
            () -> new Item(new Item.Properties()));


    public static void register(IEventBus eventBus) {
        ITEMS.register(eventBus);
    }

}

File: src/main/kotlin/auxy/paintedplanks/item/ModItems.kt
package auxy.paintedplanks.item

import net.neoforged.neoforge.client.model.generators.ItemModelBuilder
import net.neoforged.neoforge.registries.DeferredRegister
import auxy.paintedplanks.PaintedPlanks

object ModItems {
    val ITEM_REGISTRY: DeferredRegister.Items = DeferredRegister.createItems(PaintedPlanks.ID)
}
